### ISSUE TITLE
Deleting model elements produces incorrect basis set?

### ISSUE THREAD

**Issue by jebyrnes** [jslefche/piecewiseSEM#120]
Consider the keeley data where abiotic is left floating off and away with distance's effect being mediated partially via hetero on rich.

```
keeley_sem_no_abiotic <- psem(
  abiotic ~ 1,
  lm(hetero ~ distance, data=keeley),
  lm(rich ~ distance + hetero, data=keeley),
  data = keeley
)

```

The basis set *should* include abiotic ~ distance. However, it does not.

```
> basisSet(keeley_sem_no_abiotic)
$`1`
[1] "abiotic"  "hetero"   "distance"

$`2`
[1] "abiotic"  "rich"     "distance" "hetero"  
```

You can fix the `abiotic ~ distance` relationship to 0 in an `lm()`. Actually, including `lm(abiotic ~ 0*distance, data=keeley)` causes `basisSet` to bork and return a null basis set as it removes abiotic from the whole shebang.

**Comment by jslefche** [comment]
Good point. Could be solved add.claims in #127....could also just have `abiotic ~ 0*distance` as a formula in the model where that independence claim is evaluated
